sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -112,7 +112,12 @@ class SimpleFunctionRegistry extends FunctionRegistry {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    try {
+      func(children)
+    } catch {
+      case _: ClassCastException =>
```

Comment:
This is not executing the function, but just finding the function. Will it trigger ClassCastException that occurred in a function body of the given UDF? Could you give me an example?